CHANGES.md

```diff
@@ -1,5 +1,65 @@
 # Changelog
 
+## 6.1 (2023-11-03)
+[View detailed changelog](https://github.com/spinalcordtoolbox/spinalcordtoolbox/compare/6.0...6.1)
```

Comment:
Note: This changelog will still need a line summarizing the changes from #4250. But, I thought I'd get the ball rolling a little early, just to review the bulk of the changelog early, rather than waiting for everything to get merged sequentially.